File: JavaSource/src/main/java/ru/akhcheck/patterns/chainresponsibility/handlers/LogLevel.java
package ru.akhcheck.patterns.chainresponsibility.handlers;

import java.util.EnumSet;
import java.util.Set;

/** Уровни записи. Каждое звено берёт себе только нужные. */
public enum LogLevel {
    DEBUG,
    INFO,
    WARNING,
    ERROR;

    public static Set<LogLevel> all() {
        return EnumSet.allOf(LogLevel.class);
    }

    public static Set<LogLevel> problems() {
        return EnumSet.of(WARNING, ERROR);
    }
}

File: ChainResponsibility/python-source/handlers/LogLevel.py
from __future__ import annotations

from enum import Enum


class LogLevel(Enum):
    """Уровни записи. Каждый обработчик берёт себе только нужные."""

    DEBUG = "debug"
    INFO = "info"
    WARNING = "warning"
    ERROR = "error"

    @classmethod
    def all(cls) -> set[LogLevel]:
        return set(cls)

    @classmethod
    def problems(cls) -> set[LogLevel]:
        return {cls.WARNING, cls.ERROR}
